async-profiler-context/src/main/java/io/pyroscope/labels/io/pyroscope/PyroscopeAsyncProfiler.java

```diff
@@ -124,4 +128,22 @@ private static String targetLibraryFileName(final String libraryFileName) throws
 
         return libraryFileName.substring(0, libraryFileName.length() - 3) + "-" + checksum + ".so";
     }
+
+    private static boolean isMusl() {
+        try {
+            Process pr = new ProcessBuilder("ldd")
```

Comment:
slightly concerned some systems might not have ldd at all (e.g some slimmed down docker image)

Reply:
I guess this seems to be a standard way of doing this in java ecosystem https://sourcegraph.com/search?q=context:global+isMusl+lang:java+&patternType=literal